src/components/Annotation/Annotation.js

```diff
@@ -71,44 +72,51 @@ class Annotation extends React.PureComponent<Props> {
         };
 
         return (
-            <div
-                className={classNames(`ba-annotation ${className}`, {
-                    'ba-is-pending': isPending || error
-                })}
-                onBlur={onBlur}
-                onFocus={onFocus}
-            >
-                <div className='ba-annotation-content'>
-                    <Avatar className='ba-annotation-avatar' {...createdBy} />
-                    <div className='ba-annotation-text'>
-                        <div className='ba-annotation-headline'>
-                            <UserLink className='ba-annotation-user-name' {...annotator} />
-                            <Tooltip
-                                text={
-                                    <FormattedMessage
-                                        {...messages.annotationPostedFullDateTime}
-                                        values={{ time: createdAtTimestamp }}
+            <IntlProvider locale={locale} textComponent={React.Fragment}>
+                <div
+                    className={classNames(`ba-annotation ${className}`, {
+                        'ba-is-pending': isPending || error
+                    })}
+                    onBlur={onBlur}
+                    onFocus={onFocus}
+                >
+                    <div className='ba-annotation-content'>
+                        <Avatar className='ba-annotation-avatar' {...createdBy} />
+                        <div className='ba-annotation-text'>
+                            <div className='ba-annotation-headline'>
+                                <UserLink className='ba-annotation-user-name' {...annotator} />
+                                {createdAt && (
+                                    <Tooltip
+                                        text={
+                                            <FormattedMessage
+                                                {...messages.annotationPostedFullDateTime}
+                                                values={{ time: new Date(createdAt).getTime() }}
```

Comment:
Since you're using `new Date(createdAt).getTime() `, cache the result and use it here and below, instead of invoking it each time.